Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,251 @@
+        fail_test "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+    fi
+    LogMsg "Version matches between bin version and install log version"
+
+    LogMsg "Install tests passed successfully"
+}
+
+function verify_uninstall_da() {
+    LogMsg "Starting uninstall tests"
+
+    eval $uninstaller
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Uninstall failed with exit code ${ret}"
+    else
+        LogMsg "Uninstalled with exit code 0"
+    fi
+
+    directory_exists=$(verify_directory_exists)
+    if [ "directory_exists" == "true" ]; then
+        fail_test "Directory exists"
+    fi
+
+    LogMsg "Uninstall tests passed successfully"
+}
+
+function generate_network_activity() {
+    LogMsg "Generate network activity"
+    watch -n 5 curl https://microsoft.com &>/dev/null &
+    network_activity_generator_pid=$!
+}
+
+function enable_disable_da(){
+    LogMsg "Starting Enable/Disable DA tests"
+
+    verify_install_da
+
+    # Wait for 30s for DA to start running and check for service.log
+    sleep 30
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log"
+
+    # Check for service.log.1 file and confirm service.log.2 doesn't exist
+    service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
+    time_limit=$(($service_log_time + 60))
+    if [ "$DISTRO" == "debian"*] || ["$DISTRO" == "ubuntu"* ]; then
+        time_limit=$(($time_limit + 60))
+    fi
+
+    while [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ]
+    do
+        current_time=$(date +%s)
+        if [ $current_time -gt $time_limit]; then
+            SetTestStateFailed
+            return
+        fi
+        sleep 5
+    done
+
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log.1"
+    [ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && fail_test "/var/opt/microsoft/dependency-agent/log/service.log.2 exist."
+    [ ! -c "/dev/msda" ] && fail_test "/dev/msda does not exist."
+
+    # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
+    ! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && fail_test "driver setup status=0 not found"
+    ! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && fail_test starting the dependency agent not found
+
+    # Wait for DA to be running and verify by checking for MicrosoftDependencyAgent.log
+    sleep 60
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log"
+    verify_da_pid
+
+    # Wait for events and verify by checking for /var/opt/microsoft/dependency-agent/storage/*.bb files
+    sleep 120
+    ! ls /var/opt/microsoft/dependency-agent/storage/*.bb && fail_test "/var/opt/microsoft/dependency-agent/storage/*.bb does not exist."
```

Comment:
The stdout and stderr of `ls` aren't interesting, please redirect to `/dev/null`, both here and below.

Reply:
Done